cli/cmd/compose/top.go

```diff
@@ -0,0 +1,95 @@
+/*
+   Copyright 2020 Docker Compose CLI authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package compose
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+
+	"github.com/docker/compose-cli/api/client"
+)
+
+type topOptions struct {
+	*projectOptions
+}
+
+func topCommand(p *projectOptions) *cobra.Command {
+	opts := topOptions{
+		projectOptions: p,
+	}
+	topCmd := &cobra.Command{
+		Use:   "top",
+		Short: "Display the running processes",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runTop(cmd.Context(), opts, args)
+		},
+	}
+	return topCmd
+}
+
+func runTop(ctx context.Context, opts topOptions, services []string) error {
+	c, err := client.NewWithDefaultLocalBackend(ctx)
+	if err != nil {
+		return err
+	}
+	projectName, err := opts.toProjectName()
+	if err != nil {
+		return err
+	}
+	containers, err := c.ComposeService().Top(ctx, projectName, services)
+	if err != nil {
+		return err
+	}
+
+	sort.Slice(containers, func(i, j int) bool {
+		return containers[i].Name < containers[j].Name
+	})
+
+	for _, container := range containers {
+		fmt.Printf("%s\n", container.Name)
+		err := psPrinter(os.Stdout, func(w io.Writer) {
+			for _, proc := range container.Processes {
+				info := []interface{}{}
+				for _, p := range proc {
+					info = append(info, p)
+				}
+				_, _ = fmt.Fprintf(w, strings.Repeat("%s\t", len(info))+"\n", info...)
+
+			}
+			fmt.Fprintln(w)
+		},
+			container.Titles...)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
You could extract this in a function to unit test it (will need a variable for stdout)